src/test/java/org/springframework/integration/aws/inbound/S3InboundChannelAdapterTests.java

```diff
@@ -191,7 +191,7 @@ public S3InboundFileSynchronizer s3InboundFileSynchronizer() {
 			synchronizer.setPreserveTimestamp(true);
 			synchronizer.setRemoteDirectory(S3_BUCKET);
 			synchronizer.setFilter(new S3RegexPatternFileListFilter(".*\\.test$"));
-			Expression expression = PARSER.parseExpression("#this.toUpperCase() + '.a'");
+			Expression expression = PARSER.parseExpression("(#this.contains('/') ? #this.substring(#this.lastIndexOf('/') + 1) : #this).toUpperCase() + '.a'");
```

Comment:
Well, I think we can come up with `1.0.1` release over a couple weeks. And I'm sure we will have some release for SI core `4.3.x` (or even `4.2.x`) the same time.
So, you will be able to upgrade smoothly and have all the functionality aboard.